robot/impl/resource_manager_test.go

```diff
@@ -2060,3 +2062,135 @@ func TestReconfigureParity(t *testing.T) {
 		testReconfigureParity(t, files[i], files[j])
 	}
 }
+
+// Consider a case where a main part viam-server is configured with a remote part viam-server. We
+// want to ensure that once calling `ResourceNames` on the main part returns some remote resource --
+// that remote resource will always be returned until it is configured away. When either the remote
+// robot removes it from its config. Or when the main part removes the remote.
+func TestOfflineRemoteResources(t *testing.T) {
```

Comment:
full credit to @dgottlieb for the test - I ported it from [here](https://github.com/viamrobotics/rdk/pull/4268/files#diff-f60fc288e7fc9c4813d48d3c54b37fafcb3278e54f4b11704b4423b83b58f825R2077-R2207)